### PR TITLE
Export UseGetReturn and UseMutateReturn types from index

### PR DESCRIPTION
**Pull request by South-Paw**
# Why

Our app uses `UseGetReturn` and `UseMutateReturn` types but after `9.8.0` we were not longer able to import them without eslint errors. I've added them to the index exports which seems to have fixed it when locally tested in our app.

Also organised the `package.json` and added a few more checks to the `tsconfig.json`

### CODE REVIEW

File: src/scripts/ibm-openapi-validator.d.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ declare module "ibm-openapi-validator" {
    * @param openApiDoc An object that represents an OpenAPI document.
    * @param defaultMode If set to true, the validator will ignore the .validaterc file and will use the configuration defaults.
    */
-  function validator(openApiDoc: any, defaultMode = false): Promise<ValidatorResults>;
+  function validator(openApiDoc: any, defaultMode?: any): Promise<ValidatorResults>;
```
Comment (fabien0102) [inline src/scripts/ibm-openapi-validator.d.ts]: Even if I need to remove this validator at some point (way to strict to be usable 😕), I'm curious about this change!

**Comment by fabien0102** [comment]
Looks good, but I need to try locally ;)

File: src/scripts/ibm-openapi-validator.d.ts (hunk 1 above)
Comment (fabien0102) [reply src/scripts/ibm-openapi-validator.d.ts]: I reverted the change, and nothing red so far, can you revert this?

Comment (South-Paw) [reply src/scripts/ibm-openapi-validator.d.ts]: I changed it back, weird that its no longer red 🤔

Only reason I fiddled with it was because it was red 😄

**Comment by South-Paw** [comment]
@fabien0102 reverted that ibm validator type change 👍

File: package.json [hunk 2]
```diff
@@ -45,7 +45,6 @@
     "build": "run-p build:*",
     "build:project": "tsdx build",
     "build:bin": "tsc && rollup -c rollup.config.js",
-    "prepare": "tsdx build",
```
Comment (South-Paw) [inline package.json]: @fabien0102 I think this caught out whoever did the publish of `9.8.0` so I've removed it from the scripts.

See: #222

Comment (fabien0102) [reply package.json]: Correct, but this should be `npm run build` instead, so we are sure to not forget to build before publishing (oh yes, this happens so easily! 😁)

I will this directly on master before publishing 😉

**Comment by fabien0102** [comment]
@South-Paw Published on `restful-react@9.9.0` 😉

**Pull request merged**